### ISSUE TITLE
NaN in multi-node training

### ISSUE THREAD

**Issue by bpucla** [volcengine/verl#311]
NaN appears after one update (see the figure below) but the same training command with 1 node has no issue. Any suggestions?

<img width="1648" alt="Image" src="https://github.com/user-attachments/assets/7b31e7c3-d573-451a-a9c1-508b96b5ece3" />

```
python3 -m verl.trainer.main_ppo \
    data.train_files=$HOME/data/math/train.parquet \
    data.val_files=/$HOME/data/math/train.parquet \
    data.train_batch_size=1024 \
    data.val_batch_size=1312 \
    data.max_prompt_length=512 \
    data.max_response_length=1024 \
    data.return_raw_chat=True \
    actor_rollout_ref.model.path=Qwen/Qwen2.5-14B \
    actor_rollout_ref.actor.optim.lr=1e-6 \
    actor_rollout_ref.model.use_remove_padding=True \
    actor_rollout_ref.actor.ppo_mini_batch_size=256 \
    actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=16 \
    actor_rollout_ref.actor.fsdp_config.param_offload=False \
    actor_rollout_ref.actor.fsdp_config.grad_offload=False \
    actor_rollout_ref.actor.fsdp_config.optimizer_offload=False \
    actor_rollout_ref.model.enable_gradient_checkpointing=True \
    actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=32 \
    actor_rollout_ref.rollout.tensor_model_parallel_size=4 \
    actor_rollout_ref.rollout.name=vllm \
    actor_rollout_ref.rollout.gpu_memory_utilization=0.4 \
    actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=32 \
    actor_rollout_ref.ref.fsdp_config.param_offload=True \
    critic.optim.lr=1e-5 \
    critic.model.use_remove_padding=True \
    critic.model.path=Qwen/Qwen2.5-14B \
    critic.model.enable_gradient_checkpointing=True \
    critic.ppo_micro_batch_size_per_gpu=32 \
    critic.model.fsdp_config.param_offload=False \
    critic.model.fsdp_config.grad_offload=False \
    critic.model.fsdp_config.optimizer_offload=False \
    algorithm.kl_ctrl.kl_coef=0.001 \
    trainer.critic_warmup=0 \
    trainer.logger=['console','wandb'] \
    trainer.project_name='my_rl_exps' \
    trainer.experiment_name=test \
    trainer.n_gpus_per_node=8 \
    trainer.nnodes=2 \
    trainer.save_freq=50 \
    trainer.test_freq=10 \
    trainer.total_epochs=15 
```

**Comment by echo-valor** [comment]
What does the complete multi-node script look like here? Isn't this a single-node script?

**Comment by Gxy-2001** [comment]
same Nan in single node
verl0.2 + vllm0.7.2

**Comment by ChaosCodes** [comment]
same NaN, especially when batch_size is large

**Comment by none0663** [comment]
How to run  the complete multi-node scripts? Should I run the same script in the both machine?

**Comment by SauronLee** [comment]
Here is my solution https://github.com/volcengine/verl/issues/315#issuecomment-2670699073

**Comment by Life-0-1** [comment]
same issue

**Comment by Kirrito-k423** [comment]
中文：

这个问题目前还缺少足够信息，暂时不建议关闭。现有命令只展示了 `main_ppo` 的训练参数，但没有完整的多节点启动方式。当前示例通常是在 head 节点启动 Ray、等待所有 worker 加入后，只在 head 节点运行 `python3 -m verl.trainer.main_ppo`；worker 节点只需要 `ray start --address ...` 加入集群。

为了继续定位 NaN，请补充：
1. verl commit/version、vLLM/Ray/PyTorch/CUDA/NCCL 版本；
2. head/worker 上完整的 `ray start` 命令，以及是否只在 head 节点运行了 `main_ppo`；
3. `ray status` 输出和 NaN 首次出现前后的完整 trainer 日志；
4. W&B/console 中第一个变成 NaN 的具体指标名，例如 actor loss、critic loss、reward、advantage、log_prob、grad_norm 等；
5. 降低 `data.train_batch_size`、`ppo_mini_batch_size` 或 `tensor_model_parallel_size` 后是否仍复现；
6. 当前 main 分支或最新 release 是否仍能复现。

English:

There is not enough information to close this issue yet. The current report shows the `main_ppo` training arguments, but not the complete multi-node launch procedure. Current examples generally start Ray on the head node, wait until all worker nodes join the Ray cluster, and run `python3 -m verl.trainer.main_ppo` only on the head node; worker nodes only run `ray start --address ...`.

To continue debugging the NaN, could you provide:
1. verl commit/version and vLLM/Ray/PyTorch/CUDA/NCCL versions;
2. the full `ray start` commands on head/worker nodes, and whether `main_ppo` was launched only on the head node;
3. `ray status` output and the full trainer logs around the first NaN;
4. the first metric that becomes NaN in W&B/console, such as actor loss, critic loss, reward, advantage, log_prob, or grad_norm;
5. whether reducing `data.train_batch_size`, `ppo_mini_batch_size`, or `tensor_model_parallel_size` changes the behavior;
6. whether this still reproduces on current main or the latest release.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
